### PR TITLE
Several small improvements/typos for Glyphs PLUS additions for better…

### PR DESCRIPTION
**Pull request by fab-kru**
… ARIA labels.

Especially the ARIA labels might be up for debate (i.e. how do you call it when you deactivate the Like button - I went with "Undo Like"), but we need some kind of improvement here or people won’t know what’s happening.

The rest should not be too controversial, or so I hope. I still need more practice with this, so any change for the worse not intended. Feel free to cherrypick or come back to me with questions/suggestions.

### CODE REVIEW

File: src/UI/Component/Symbol/Factory.php [hunk 1]
```diff
@@ -15,17 +15,17 @@ interface Factory
 	 * ---
 	 * description:
 	 *   purpose: >
-	 *     Icons are quickly comprehensible and recognizable graphics.
-	 *     They indicate the functionality or nature of a text-element or context:
+	 *     Icons are quickly comprehensible and recognizable graphics that MUST be used with a text label.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Factory.php:18]: Do not state rules in the descriptions section. Please remove. If you are sure this rule is needed, move to the rules/usage section. Note that an additional rule would need a JF decision, so better remove in this PR.

File: src/UI/Component/Symbol/Factory.php [hunk 2]
```diff
@@ -62,19 +62,17 @@ public function icon(): Icon\Factory;
 	 * description:
 	 *   purpose: >
 	 *       Glyphs map a generally known concept or symbol to a specific concept in ILIAS.
-	 *       Glyphs are used when space is scarce.
+	 *       Glyphs SHOULD NOT be used with a text label. They are used when space is scarce.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Factory.php:65]: Do not state rules in the descriptions section. Please remove. If you are sure this rule is needed, move to the rules/usage section. Note that an additional rule would need a JF decision, so better remove in this PR.

File: src/UI/Component/Symbol/Factory.php [hunk 3]
```diff
@@ -62,19 +62,17 @@ public function icon(): Icon\Factory;
...
 	 *   composition: >
-	 *       A glyph is a typographical character that represents
-	 *       something else. As any other typographical character, they can be
-	 *       manipulated by regular CSS. If hovered they change their background
-	 *       to indicate possible interactions.
+	 *       A glyph is a typographical character. As any other typographical character, they can be
+	 *       manipulated by regular CSS. If hovered, they MUST change either their color OR their background-color in order to indicate possible interactions.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Factory.php:68]: Do not state rules in the descriptions section. Please remove. If you are sure this rule is needed, move to the rules/usage section. Note that an additional rule would need a JF decision, so better remove in this PR.

File: src/UI/Component/Symbol/Factory.php [hunk 4]
```diff
@@ -89,21 +87,21 @@ public function icon(): Icon\Factory;
 	 *   usage:
 	 *       1: Glyphs MUST NOT be used in content titles.
 	 *       2: >
-	 *          Glyphs MUST be used for cross-sectional functionality such as mail for
+	 *          Glyphs MUST be used for cross-sectional functionality such as Mail for
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Factory.php:90]: Why is mail capitalised here? We only capitalise names of other UI Components in the description and rules, to make them stand out.

File: src/UI/Component/Symbol/Factory.php [hunk 5]
```diff
@@ -89,21 +87,21 @@ public function icon(): Icon\Factory;
...
 	 *          example and NOT for representing objects.
 	 *       3: >
 	 *          Glyphs SHOULD be used for very simple tasks that are repeated at
 	 *          many places throughout the system.
 	 *       4: >
-	 *          Services such as mail MAY be represented by a glyph AND an icon.
+	 *          Services such as Mail MAY be represented either by a glyph OR by an icon plus text label, depending on the usage scenario.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Factory.php:96]: Why is mail capitalised here? We only capitalise names of other UI Components in the description and rules, to make them stand out. Note, Glyph and Icon should be capitalised, since they are other UI Components, not your bad ;-)

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 6]
```diff
@@ -13,15 +13,14 @@ interface Factory {
 	 * ---
 	 * description:
 	 *   purpose: >
-	 *       The Settings Glyph triggers opening a Dropdown to edit settings of the displayed block.
+	 *       The Settings Glyph triggers the opening of a dropdown that allows to edit settings of the displayed block.
 	 *   composition: >
 	 *       The Settings Glyph uses the glyphicon-cog.
 	 *   effect: >
-	 *      Upon clicking a settings Dropdown is opened.
+	 *      Upon clicking, a settings dropdown is opened.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:20]: Dropdown is a UI Component => capitalise

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 7]
```diff
@@ -526,7 +521,7 @@ public function comment(string $action = null): Glyph;
 	 *          A Status Counter MUST indicate the overall amount of like expressions.
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be 'Like'.
+	 *          The aria-label MUST be ‘Like' for the first (inactive) version of the glyph and ‘Undo Like' for the second (active) version.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:524]: There is a whole set of such accessibility rule change for the aria-label. I like them. 

They need JF decision, and they will also need some additional implementation. Make a separate PR in which you pack all those aria-rule changes, which we then propose to the JF.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 8]
```diff
@@ -552,7 +547,7 @@ public function like(string $action = null): Glyph;
 	 *          A Status Counter MUST indicate the overall amount of love expressions.
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be 'Love'.
+	 *          The aria-label MUST be ‘Love' for the first (inactive) version of the glyph and ‘Undo Love' for the second (active) version.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:550]: Separate PR, see above.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 9]
```diff
@@ -578,7 +573,7 @@ public function love(string $action = null): Glyph;
 	 *          A Status Counter MUST indicate the overall amount of dislike expressions.
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be 'Dislike'.
+	 *          The aria-label MUST be ‘Dislike' for the first (inactive) version of the glyph and ‘Undo Dislike' for the second (active) version.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:576]: Separate PR, see above.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 10]
```diff
@@ -604,7 +599,7 @@ public function dislike(string $action = null): Glyph;
 	 *          A Status Counter MUST indicate the overall amount of laugh expressions.
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be 'Laugh'.
+	 *          The aria-label MUST be ‘Laugh' for the first (inactive) version of the glyph and ‘Undo Laugh' for the second (active) version.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:602]: Separate PR, see above.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 11]
```diff
@@ -630,7 +625,7 @@ public function laugh(string $action = null): Glyph;
 	 *          A Status Counter MUST indicate the overall amount of astounded expressions.
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be 'Astounded'.
+	 *          The aria-label MUST be ‘Astounded' for the first (inactive) version of the glyph and ‘Undo Astounded' for the second (active) version.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:628]: Separate PR, see above.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 12]
```diff
@@ -656,7 +651,7 @@ public function astounded(string $action = null): Glyph;
 	 *          A Status Counter MUST indicate the overall amount of sad expressions.
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be 'Sad'.
+	 *          The aria-label MUST be ‘Sad' for the first (inactive) version of the glyph and ‘Undo Sad' for the second (active) version.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:654]: Separate PR, see above.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 13]
```diff
@@ -682,7 +677,7 @@ public function sad(string $action = null): Glyph;
 	 *          A Status Counter MUST indicate the overall amount of angry expressions.
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be 'Angry'.
+	 *          The aria-label MUST be ‘Angry' for the first (inactive) version of the glyph and ‘Undo Angry' for the second (active) version.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:680]: Separate PR, see above.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 14]
```diff
@@ -710,7 +705,7 @@ public function angry(string $action = null): Glyph;
 	 *          The Eye Closed Glyph MUST only be used with Password-Inputs.
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be "eye closed - click to hide the input's contents".
+	 *          The aria-label MUST be "Click to hide the password".
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:708]: Separate PR, see above.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 15]
```diff
@@ -737,7 +732,7 @@ public function eyeclosed(string $action = null): Glyph;
 	 *          The Eye Open Glyph MUST only be used with Password-Inputs.
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be "eye open - click to reveal the input's contents".
+	 *          The aria-label MUST be "Click to reveal the password".
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:735]: Separate PR, see above.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 16]
```diff
@@ -790,14 +785,14 @@ public function attachment(string $action = null): Glyph;
...
-	 *          If there are no changes to reset, the Reset Glyph MUST be deactivated (or not clickable).
+	 *          If there are no changes to reset, the Reset Glyph MUST be deactivated (or not be clickable).
 	 *   style:
 	 *       1: >
 	 *          The deactivated state of the Reset Glyph MUST be visually noticeable for the user, i.e. by
 	 *          greying out the Reset Glyph.
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be 'Reset'.
+	 *          The aria-label MUST be 'Reset'. The deactivated state of the Glyph MUST have the aria-label 'Reset not possible'.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:795]: Separate PR, see above.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 17]
```diff
@@ -824,14 +819,14 @@ public function reset(string $action = null): Glyph;
...
-	 *          If there are no changes to apply, the Apply Glyph MUST be deactivated (or not clickable).
+	 *          If there are no changes to apply, the Apply Glyph MUST be deactivated (or not be clickable).
 	 *   style:
 	 *       1: >
 	 *          The deactivated state of the Apply Glyph MUST be visually noticeable for the user, i.e. by greying out
 	 *          the Apply Glyph.
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be 'Apply'.
+	 *          The aria-label MUST be 'Apply'. The deactivated state of the Glyph MUST have the aria-label 'Applying not possible'.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:829]: Separate PR, see above.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 18]
```diff
@@ -897,15 +892,15 @@ public function help(string $action = null): Glyph;
 	 *   composition: >
 	 *      The Calendar Glyph uses the glyphicon-calendar.
 	 *   effect: >
-	 *       Clicking the calendar glyph will usually open a date-picker.
+	 *       Clicking the calendar glyph SHOULD open a date-picker.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:895]: no rules in the description section. Leave as is.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 19]
```diff
@@ -897,15 +892,15 @@ public function help(string $action = null): Glyph;
...
 	 *
 	 * context:
 	 *    - Use in conjunction with Date-Inputs.
 	 *
 	 * rules:
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be 'Calendar'.
+	 *          The aria-label MUST be 'Pick date' for opening date-pickers. It MUST be adapted for other use cases.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:903]: Separate PR, see above.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 20]
```diff
@@ -920,15 +915,15 @@ public function calendar($action = null);
 	 *   composition: >
 	 *      The Time Glyph uses the glyphicon-time.
 	 *   effect: >
-	 *       Clicking the time glyph will usually open a time-picker.
+	 *       Clicking the time glyph SHOULD open a time-picker.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:918]: no rules in the description section. Leave as is.

File: src/UI/Component/Symbol/Glyph/Factory.php [hunk 21]
```diff
@@ -920,15 +915,15 @@ public function calendar($action = null);
...
 	 *
 	 * context:
 	 *    - Use in conjunction with Date-Inputs.
 	 *
 	 * rules:
 	 *   accessibility:
 	 *       1: >
-	 *          The aria-label MUST be 'Time'.
+	 *          The aria-label MUST be 'Pick time' for opening time-pickers. It MUST be adapted for other use cases.
```
Comment (Amstutz) [inline src/UI/Component/Symbol/Glyph/Factory.php:926]: Separate PR, see above.

**Review by Amstutz — COMMENTED** [review]
Hi @fab-kru 

Thank you very much for your great effort! Most changes are fine and are of great help to enhance the quality of the description. For this PR, I left some inline comments to quickly direct you to the relevant places.

Along with some details (see comments) two things need to be done, before I can merge this contribution:
- [ ] In several cases, you mixed rules with description: Do not use key words such RFC 2119 Requirement Keywords (SHOULD, MUST etc.) in the description sections. Only use them in the rules. We want to clearly separate between prose description and testable rules.
- [ ] Move the accessabilty aria-rule changes: Move those changes to an own PR. Those changes need to be accepted by the JF and might lead to discussion. It's easier to discuss them, if they are factored out in their own PR. Note that I like your changes. Also note, that they will spawn bug reports as soon as accepted, see: https://mantis.ilias.de/view.php?id=25803

Ping me as soon as done, and I will merge your changes. Thx a lot!

**Comment by klees** [comment]
@Amstutz @fab-kru Do you need me here?

**Comment by Amstutz** [comment]
@klees, nope, your opinion is always welcome however, not strictly required in this case...

**Comment by Amstutz** [comment]
I close this down for the moment, parts of it are already merged (see: #2038). Please open a new one a soon as ready.
